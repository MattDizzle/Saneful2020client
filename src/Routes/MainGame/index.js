import React, { useContext } from 'react';
import { useHistory } from "react-router-dom";
import GameContext from '../../Context/GameContext';
import GameWindow from '../../Components/GameWindow/GameWindow';

import './MainGame.scss';

const MainGame = () => {
  //if the player reloads the game basically breaks so we need to bring them back to dashboard on reload
  const context = useContext(GameContext);
  const history = useHistory();

  if (context.gameData.saved_game_id === 0) {
    context.saveGame();
    history.push('/dashboard');
  }

  return (
    <section className='MainGame'>
      <GameWindow />
    </section>
  );
};

export default MainGame;