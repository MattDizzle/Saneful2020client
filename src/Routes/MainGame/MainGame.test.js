import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import MainGame from '../MainGame';


describe(`MainGame Component`, () => {
  describe(`Smoke test`, () => {
    it(`Renders without crashing`, () => {
      const div = document.createElement('div');
      ReactDOM.render(
        <BrowserRouter>
          <MainGame />
        </BrowserRouter>
        , div);
      ReactDOM.unmountComponentAtNode(div);
    });
  });

  describe(`Snapshot test`, () => {
    it(`Renders the UI as expected`, () => {
      const tree = renderer
        .create(
          <BrowserRouter>
            <MainGame />
          </BrowserRouter>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});