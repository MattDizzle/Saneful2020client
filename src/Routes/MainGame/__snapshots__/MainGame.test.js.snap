// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainGame Component Snapshot test Renders the UI as expected 1`] = `
<section
  className="MainGame"
>
  <section
    className="GameWindow"
    onKeyDown={[Function]}
    tabIndex={0}
  >
    <img
      alt="background"
      className="cells"
      src="images/grid/draft1nogrid.png"
    />
    <div
      className="cells"
    >
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="action-cell"
        onClick={[Function]}
      >
        <img
          alt="action"
          src="images/pixel-elements/exclamation.png"
        />
      </div>
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="action-cell"
        onClick={[Function]}
      >
        <img
          alt="action"
          src="images/pixel-elements/exclamation.png"
        />
      </div>
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="action-cell"
        onClick={[Function]}
      >
        <img
          alt="action"
          src="images/pixel-elements/exclamation.png"
        />
      </div>
      <div
        className="action-cell"
        onClick={[Function]}
      >
        <img
          alt="action"
          src="images/pixel-elements/exclamation.png"
        />
      </div>
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
      <div
        className="cell"
        onClick={[Function]}
      />
    </div>
    <div
      className="player-animation"
    >
      <img
        alt="player"
        className="Player"
        src="/images/characters/c1/c1right2.png"
      />
    </div>
    <div
      className="UI"
    >
      <div
        className="ui-left"
      >
        <div>
          <p>
            Day 0
          </p>
          <p>
            00:00
          </p>
        </div>
      </div>
      <div
        className="ui-right"
      >
        <div
          className="money-save"
        >
          <p>
            Money: $0
          </p>
          <button
            onClick={[Function]}
          >
            Save
          </button>
        </div>
        <div
          className="title-barContainer"
        >
          <p>
            Sanity:
          </p>
          <div
            className="barContainer"
          >
            <div
              style={
                Object {
                  "backgroundColor": "#250cc7",
                  "borderRadius": "10px",
                  "height": "100%",
                  "textAlign": "right",
                  "transition": "0.25s",
                  "width": "0%",
                }
              }
            >
              <span
                className="barLabel"
              >
                0
              </span>
            </div>
          </div>
        </div>
        <div
          className="title-barContainer"
        >
          <p>
            Health:
          </p>
          <div
            className="barContainer"
          >
            <div
              style={
                Object {
                  "backgroundColor": "#ab111b",
                  "borderRadius": "10px",
                  "height": "100%",
                  "textAlign": "right",
                  "transition": "0.25s",
                  "width": "0%",
                }
              }
            >
              <span
                className="barLabel"
              >
                0
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</section>
`;
